libbeat/docs/upgrading.asciidoc

```diff
@@ -1,160 +1,216 @@
-layout for configuration, where each module is configured in its own
-configuration file.
+Starting with 7.0, the fields exported by {beats} conform to the
+{ecs-ref}/index.html[Elastic Common Schema (ECS)]. Many of the exported fields
+have been renamed. See {beats-ref}/breaking-changes-7.0.html[Breaking
+changes in 7.0] for the full list of changed names.
 
-While the all-in-one configuration is still fully supported, we recommend moving
-to the new layout at upgrade time. This typically means starting off with the new
-default configuration and modifying it with the custom settings that you had in
-your old files.
+To help you transition to the new fields, we provide a compatibility layer in
+the form of ECS-compatible field aliases. To use the aliases, set
+`migration.6_to_7.enabled: true` in the Beat's configuration file *before* you
+run Beats 7.0 to ingest data for the first time.
 
-// TODO: link to the docs on modules.d
+The compatibility layer enables you to use 6.x dashboards and visualizations
+with indices created by {beats} 7.0 or later.
 
-===== New command `test config` command
+// TODO: Describe how to wipe out old data and overwrite the template for
+// situations where users ingest data before realizing they need to set the
+// migration flag.
+//
+// I do wonder why this flag is not the default, but maybe it creates a lot of
+// extra noise for users who don't want compatibility?
+
+==== Upgrade dashboards
+
+We recommend that you reload the {kib} dashboards after upgrading {kib} and
+{beats}. This way, you can take advantage of the new {beats} dashboards created
+for the 7.0 release. 
 
-Beats 6.0 introduces a new test command for testing the configuration file. For
-example:
+Reloading the dashboards will overwrite any manual changes that you've
+made. To avoid overwriting existing dashboards, save them under a new ID before
+proceeding.
+
+To reload the dashboards, run the `setup` command with the `--dashboards`
+option specified. For example:
 
 [source,shell]
-------------------------------------------------------------------------------
-metricbeat test config
-------------------------------------------------------------------------------
+----
+metricbeat setup --dashboards
+----
 
-The old `-configtest` flag is still available, but deprecated.
+Also, if you plan to run {beats} 6.7 and 7.0 in parallel, set
+`migration.6_to_7.enabled: true` to make 6.x dashboards compatible with 7.0
+{beats} indices *before* you start ingesting data. For more information, see
+<<enable-ecs-compatibility>>.
 
-===== Reference configuration files
+==== Configure {ls} to use versioned templates and indices
 
-The `<beatname>.full.yml` file, which contains all the non-deprecated
-configuration options is renamed to `<beatname>.reference.yml` starting with
-Beats 6.0. We recommend using this file as a reference only. It's not intended
-to be used in production.
+Before upgrading to 7.0, make sure {ls} uses versioned templates and indices. 
```

Comment:
I know it's implied, but maybe it's worth explicitly stating, "If you're using Logstash to ingest your Beats data..." so that people won't think LS is _required_ for upgrading.

Reply:
Also, what does "versioned templates" mean in this case? The link below goes to [plugins-inputs-beats.html](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-beats.html) which currently only mentions versioned indices.

Reply:
CC @ruflin ^

Reply:
I'm modifying this section a bit to show the settings (including manage_template => false, which let's Beats manage the template).